src/Symfony/Component/Console/Event/ConsoleCommandEvent.php

```diff
@@ -59,4 +64,133 @@ public function commandShouldRun()
     {
         return $this->commandShouldRun;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInput()
+    {
+        if ($this->inputDecorator) {
+            return $this->inputDecorator;
+        }
+
+        return $this->inputDecorator = new InputDecorator(parent::getInput());
+    }
+}
+
+/**
+ * @internal
+ */
+final class InputDecorator implements InputInterface
+{
+    private $inner;
+    private $overriddenArguments = array();
+    private $overriddenOptions = array();
+
+    public function __construct(InputInterface $inner)
+    {
+        $this->inner = $inner;
+    }
+
+    public function __call($method, $arguments)
+    {
+        if ('setArgument' === $method) {
+            return $this->setArgument($arguments[0], $arguments[1]);
+        }
+
+        if ('setOption' === $method) {
+            return $this->setOption($arguments[0], $arguments[1]);
+        }
```

Comment:
These are not neeeded as `__call` isn't called on defined methods. Btw, not sure if this `__call()` function is needed at all, given all input interface methods are implemented directly.